benchmarks/nyc_taxi/run_local.py

```diff
@@ -0,0 +1,78 @@
+"""Run local version of Bodo, Dask, Modin, and Pyspark benchmarks
+Accepts a list of files to use as the High Volume For Hire Vehicle dataset
+(defaults to first 5 million rows of the dataset). For exhaustive list of
+files, see s3://bodo-example-data/nyc-taxi/fhvhv. You can also optionally
+specify the system to run e.g. just run bodo or "all" to run on all systems.
+
+usage:
+    python run_local.py --dataset FILE --system SYSTEM
+"""
+
+import argparse
+
+from .bodo.nyc_taxi_precipitation import (
+    get_monthly_travels_weather as bodo_get_monthly_travels_weather,
+)
+from .dask.nyc_taxi_precipitation import (
+    local_get_monthly_travels_weather as dask_get_monthly_travels_weather,
+)
+from .modin_ray.nyc_taxi_precipitation import (
+    local_get_monthly_travels_weather as modin_get_monthly_travels_weather,
+)
+from .spark.spark_nyc_taxi_precipitation import (
+    get_monthly_travels_weather as spark_get_monthly_travels_weather,
+)
+
+hvfhv_small = "s3://bodo-example-data/nyc-taxi/fhvhv_5M_rows.pq"
+
+
+def get_spark_paths(files: str | list[str]) -> str:
+    """Gets the equivalent path for spark to use."""
+    if isinstance(files, str):
+        return files.replace("s3://", "s3a://").replace("fhvhv/", "fhvhv-rewrite/")
+
+    assert len(files) != 1, "Spark benchmark expects a single path argument."
+
+    return files[0].replace("s3://", "s3a://").replace("fhvhv/", "fhvhv-rewrite/")
```

Comment:
PySpark has slightly different syntax for reading multiple parquet files so we'd have to modify the script to allow this. I think it's fine to just cap it at one since it can't handle more than one parquet file from this dataset anyways (runs out of memory).